Mathlib/CategoryTheory/Adjunction/Lifting/Right.lean

```diff
@@ -0,0 +1,239 @@
+/-
+Copyright (c) 2024 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Monad.Adjunction
+import Mathlib.CategoryTheory.Monad.Equalizer
+
+/-!
+# Adjoint lifting
+
+This file gives two constructions for building right adjoints: the adjoint triangle theorem and the
+adjoint lifting theorem.
+
+The adjoint triangle theorem concerns a functor `F : B ⥤ A` with a right adjoint `U` such
+that `η_X : X ⟶ UFX` is a regular mono. Then for any category `C` with equalizers of coreflexive
+pairs, a functor `L : C ⥤ B` has a right adjoint if (and only if) the composite `L ⋙ F` does.
+Note that the condition on `F` regarding `η_X` is automatically satisfied in the case when `F` is
+a comonadic functor, giving the corollary: `isLeftAdjoint_triangle_lift_comonadic`, i.e. if `F` is
+comonadic, `C` has coreflexive equalizers then `L : C ⥤ B` has a right adjoint provided `L ⋙ F`
+does.
+
+The adjoint lifting theorem says that given a commutative square of functors (up to isomorphism):
+
+      Q
+    A → B
+  U ↓   ↓ V
+    C → D
+      L
+
+where `V` is comonadic, `U` has a right adjoint and `A` has coreflexive equalizers, then if `L` has
```

Comment:
This line seems inconsistent with the other file. Based on the content (I am not familiar with the proofs) it seems that this one is correct, and the other one is unnecessarily strong? Am I right in this?